sync default branch if no local changes

blairham opened the pull request:
## Status
**READY**

## Description
This change will allow you to keep a repo current if there are no local changes. Sometimes you just want to keep the repo in sync with the remote. This will allow you to do this in bulk when it's safe.

gabrie30 commented:
If not sure if I'm understanding this feature correctly, it is to skip updating repos that have changes made to them?

I tried cloning an org, deleting a file on the main branch, and cloning again. I would expect the repo to be skipped but the deleted file was recreated.

It might help if you could explain a little bit more about the feature when it would be useful and a brief outline of what it does.

blairham commented:
> If not sure if I'm understanding this feature correctly, it is to skip updating repos that have changes made to them?
> 
> I tried cloning an org, deleting a file on the main branch of a repo, and then cloning the org again. I would expect the repo with the deleted file to be skipped but the deleted file was recreated.
> 
> It might help if you could explain a little bit more about the feature when it would be useful and a brief outline of what it does.

You're right, I added more documentation and made this much more resilient. I also added a new flag that can be turned on and off, with the default setting being off.

blairham commented:
@gabrie30 I've made some changes to incorporate the Gitter so we can call the functions of that interface. This means that, among both PRs, you should merge this one before merging the go-git branch. Once this merge is complete, I'll rebase the go-git code and then add the new Gitter functionality. By doing it this way, we avoid rewriting the sync functionality; instead, it's managed by the client that implements the Gitter interface. This makes it much more modular.

blairham commented:
I'm going to close this and re-open after I fix this after the re-work
